### DISCUSSION TITLE
Custom Error Handler

### DISCUSSION THREAD

**Discussion by sinanerkan** [Farfetch/kafkaflow#528] [Q&A]
I need to register an error handler in case of a connection loss with brokers. There's a built-in logging handler which notifies about the error occurred, but is there a way to add additional flow in case of an error?

**Comment by JoaoRodriguesGithub** [comment]
Hello @sinanerkan,

Can you please detail the scenario that you intend to do? 

What is the type of error you intent to catch? And what is your intent to react to that error? 

KafkaFlow already exposes some of events regarding the messages, but at this time we don't have available events regarding the brokers:
- [Global Events](https://farfetch.github.io/kafkaflow/docs/guides/global-events)

**Comment by sinanerkan** [comment]
Hello @JoaoRodriguesGithub 

Well, in case a connection loss, I need to do some stuff on my application until the connection is restored. To be specific I'm referring to the below part of the library.

https://github.com/Farfetch/kafkaflow/blob/master/src/KafkaFlow/Consumers/Consumer.cs

If you expose a method which will allow the users to register additional error handlers by adding our handlers to _errorsHandlers object, that would be great.

Now come to think of it, we need to be able to register statistics handlers as well to be able to reverse the operation that is registered on error handler.
